File: mobile/lib/features/reimbursement/domain/entities/reimbursement.dart
import 'package:equatable/equatable.dart';

class Reimbursement extends Equatable {
  final String id;
  final String employeeId;
  final String title;
  final String? description;
  final double amount;
  final String? attachmentUrl;
  final String status;
  final String? rejectedReason;
  final DateTime createdAt;
  final String? employeeName;

  const Reimbursement({
    required this.id,
    required this.employeeId,
    required this.title,
    this.description,
    required this.amount,
    this.attachmentUrl,
    required this.status,
    this.rejectedReason,
    required this.createdAt,
    this.employeeName,
  });

  factory Reimbursement.fromJson(Map<String, dynamic> json) {
    return Reimbursement(
      id: json['id']?.toString() ?? json['ID']?.toString() ?? '',
      employeeId: json['employee_id']?.toString() ?? json['EmployeeID']?.toString() ?? '',
      title: json['title'] ?? json['Title'] ?? '',
      description: json['description'] ?? json['Description'],
      amount: (json['amount'] as num?)?.toDouble() ?? (json['Amount'] as num?)?.toDouble() ?? 0.0,
      attachmentUrl: json['attachment_url'] ?? json['AttachmentURL'],
      status: json['status'] ?? json['Status'] ?? 'PENDING',
      rejectedReason: json['rejected_reason'] ?? json['RejectedReason'],
      createdAt: json['created_at'] != null ? DateTime.parse(json['created_at']).toLocal() : 
                 (json['CreatedAt'] != null ? DateTime.parse(json['CreatedAt']).toLocal() : DateTime.now()),
      employeeName: json['employee_name'] ?? json['EmployeeName'],
    );
  }

  @override
  List<Object?> get props => [
        id,
        employeeId,
        title,
        description,
        amount,
        attachmentUrl,
        status,
        rejectedReason,
        createdAt,
        employeeName,
      ];
}

File: backend/internal/domain/reimbursement.go
package domain

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type ReimbursementStatus string

const (
	ReimbursementPending  ReimbursementStatus = "PENDING"
	ReimbursementApproved ReimbursementStatus = "APPROVED"
	ReimbursementRejected ReimbursementStatus = "REJECTED"
)

type Reimbursement struct {
	ID             uuid.UUID           `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	EmployeeID     uuid.UUID           `gorm:"type:uuid;not null" json:"employee_id"`
	Title          string              `gorm:"type:varchar(255);not null" json:"title"`
	Description    string              `gorm:"type:text" json:"description"`
	Amount         float64             `gorm:"type:decimal(15,2);not null" json:"amount"`
	AttachmentURL  string              `gorm:"type:text" json:"attachment_url"`
	Status         ReimbursementStatus `gorm:"type:varchar(20);default:'PENDING'" json:"status"`
	ApprovedBy     *uuid.UUID          `gorm:"type:uuid" json:"approved_by"`
	RejectedReason *string             `gorm:"type:text" json:"rejected_reason"`
	CreatedAt      time.Time           `gorm:"default:now()" json:"created_at"`
	UpdatedAt      time.Time           `gorm:"default:now()" json:"updated_at"`

	Employee *Employee `gorm:"foreignKey:EmployeeID;constraint:OnDelete:CASCADE;" json:"employee,omitempty"`
	Approver *Employee `gorm:"foreignKey:ApprovedBy;constraint:OnDelete:SET NULL;" json:"approver,omitempty"`
}

func (r *Reimbursement) BeforeCreate(tx *gorm.DB) (err error) {
	if r.ID == uuid.Nil {
		r.ID = uuid.New()
	}
	return
}
